src/Helpers/EventHelper.ts

```diff
@@ -0,0 +1,159 @@
+            if (creepIsSquadMember) {
+                // Find flag for the creep in a squad
+                switch (flagType) {
+
+                    case STANDARD_SQUAD:
+                        if (STANDARD_SQUAD_ARRAY.includes(creepRole)) {
+                            requestingFlag = attackFlag;
+                        }
+                        break;
+                }
+            }
+            else {
+                // Find flag for creep NOT in a squad (solo)
+                switch (flagType) {
+
+                    case ZEALOT_SOLO:
+                        if (ZEALOT_SOLO_ARRAY.includes(creepRole)) {
+                            requestingFlag = attackFlag;
+                        }
+                        break;
+
+                    // If
+                    case STALKER_SOLO:
+                        if (STALKER_SOLO_ARRAY.includes(creepRole)) {
+                            requestingFlag = attackFlag;
+                        }
+                        break;
+                }
+            }
+
+            // If we found the flag, break the loop
+            if (requestingFlag !== undefined) {
+                break;
+            }
+        } // -------------------------------------
+
+        // Throw an error if we didn't find the flag
+        if (requestingFlag === undefined) {
+            throw new UserException(
+                "The event for spawning a military creep was improperly handled.",
+                "The creep couldn't increment the correct flags memory, meaning the attack flag\n" +
+                "will not be removed properly and must be done manually. [ " + creep.name + " ].",
+                ERROR_ERROR
+            );
+        }
+
+        // Increment the current spawned, if thats the cut off, complete the associated real flag
+        requestingFlag!.currentSpawnCount = requestingFlag!.currentSpawnCount + 1;
+        if (requestingFlag!.currentSpawnCount >= requestingFlag!.squadSize && requestingFlag!.currentSpawnCount > 0) {
+            if (Game.flags[requestingFlag!.flagName]) {
+                Game.flags[requestingFlag!.flagName].memory.complete = true;
+            }
+        }
+    }
+
+    /**
+     * scan for structure built events
+     * @param room the room we are scanning
+     */
+    public static scanForStructureBuiltEvents(room: Room): void {
+
+        // ISSUES
+        // The structure is NOT getting set as processed
+        // Solved, defined the property for it. should be fine now
+
+        // Get all structures in the room that have not yet been processed
+        const structures: Structure[] = MemoryApi.getStructures(
+            room.name,
+            (structure: Structure) =>
+                structure.structureType !== STRUCTURE_KEEPER_LAIR
+                && structure.structureType !== STRUCTURE_CONTROLLER &&
+                (structure.memory.processed === false || structure.memory.processed === undefined),
+            true
```

Comment:
I would prefer if we don't invalidate the cache every tick.. Would it be possible to limit this?

Reply:
currently not possible to do this and have the bugs this fixes, still be fixed (requires finding things on the same tick, could only trigger invalidation of cache on a certain trigger that would indicate something like this happened (ie room events has a creep build action in it or something)